### PR TITLE
paddle frontend cholesky_solve added

### LINKED ISSUE #21182: cholesky_solve

### PR DESCRIPTION
**Pull request by dhanush-2501**
Close #21182

### CODE REVIEW

**Comment by dhanush-2501** [comment]
Hi @AnnaTz, can you pls check on this?

File: ivy/functional/frontends/paddle/tensor/linalg.py [hunk 1]
```diff
@@ -122,6 +122,16 @@ def solve(x1, x2, name=None):
     return ivy.solve(x1, x2)
 
 
+# cholesky_solve
+@with_supported_dtypes({"2.5.1 and below": ("float32", "float64")}, "paddle")
+@to_ivy_arrays_and_back
+def cholesky_solve(y, x, /, *, upper=False, name=None):
```
Comment (AnnaTz) [inline ivy/functional/frontends/paddle/tensor/linalg.py:128]: The original function puts `x` before `y`. We should do so as well; the signature should be exactly the same.

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_linalg.py [hunk 2]
```diff
@@ -528,6 +528,87 @@ def test_paddle_solve(
...
+        )
+    )
+    shared_size = draw(
+        st.shared(helpers.ints(min_value=2, max_value=4), key="shared_size")
+    )
+    return input_dtype, draw(
+        helpers.array_values(
+            dtype=input_dtype, shape=tuple([shared_size, 1]), min_value=2, max_value=5
+        )
+    )
```
Comment (AnnaTz) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_linalg.py:576]: You don't need to copy-paste other tests' helpers. It's cleaner to just import them from the file you found them in.

**Review by AnnaTz — CHANGES_REQUESTED** [review]

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_linalg.py (hunk 2 above)
Comment (dhanush-2501) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_linalg.py:576]: Actually I tried that but that function was expecting a tensor. And the tests didn't pass. Yet still i would give a try again

File: ivy/functional/frontends/paddle/tensor/linalg.py (hunk 1 above)
Comment (dhanush-2501) [reply ivy/functional/frontends/paddle/tensor/linalg.py]: Actually that was a bug in their documentation. I found that, reported and fixed that to them as well. thats the reason for that confusion. link to the paddle paddle bug : https://github.com/PaddlePaddle/Paddle/issues/55998

**Review by AnnaTz — APPROVED** [review]
Looks good, thanks!

**Pull request merged**